### PR TITLE
applications: nrf_desktop: Change settings backend to NVS

### PR DESCRIPTION
**Pull request by emob-nordic**
Replace settings backend for NVS instead of FCB.

Jira: NCSDK-9958.

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

I asked documentation lead for 1.7 release to add release note file. I would make a note about that change once the file is in. The change was briefly tested with nrf52840dk with dongle configuration.

@wiminordic do we have any tests that could verify backend for settings automatically? Could we run them nightly?

### CODE REVIEW

**Comment by pdunaj** [comment]
I think you can already add this to release notes as it is wip document until release actually happens.
https://github.com/nrfconnect/sdk-nrf/blob/master/doc/nrf/releases/release-notes-latest.rst

**Comment by pdunaj** [comment]
810 does not fit NVS or we already used it there?

**Comment by emob-nordic** [comment]
> 810 does not fit NVS or we already used it there?

We already used it before as a part of reducing flash size.

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -40,7 +40,10 @@ nRF9160
 nRF5
 ====
 
-There are no entries for this section yet.
+nRF Desktop
+-----------
+
+* Settings backed changed from FCB to NVS.
```
Comment (MarekPieta) [inline doc/nrf/releases/release-notes-latest.rst:46]: `backend`.

**Review by MarekPieta — APPROVED** [review]
Fix typo in release notes please

**Review by pdunaj — APPROVED** [review]
Ok - apart from the typo

**Comment by jfischer-no** [comment]
Is that ready?

**Comment by emob-nordic** [comment]
> Is that ready?

Let me test one more thing and will ask you in DM if can be merged.

**Pull request merged**